public/app/features/dashboard/components/PanelEditor/getVisualizationOptions.tsx

```diff
@@ -97,16 +97,14 @@ export function getVisualizationOptions(props: OptionPaneRenderProps): OptionsPa
    * Field options
    */
   for (const fieldOption of plugin.fieldConfigRegistry.list()) {
-    if (
-      fieldOption.isCustom &&
-      fieldOption.showIf &&
-      !fieldOption.showIf(currentFieldConfig.defaults.custom, data?.series)
-    ) {
-      continue;
-    }
-
-    if (fieldOption.hideFromDefaults) {
-      continue;
+    if (fieldOption.isCustom) {
+      if (fieldOption.showIf && !fieldOption.showIf(currentFieldConfig.defaults.custom, data?.series)) {
+        continue;
+      }
+    } else {
+      if (fieldOption.showIf && !fieldOption.showIf(currentFieldConfig.defaults, data?.series)) {
+        continue;
+      }
```

Comment:
This is a bit funky. `showIf` receiving the custom property if the option is party of the `custom` property, and the field config if it's not.

Also, it might be nicer to have this option disabled in the UI when both min and max are set, rather than having it hidden.

Reply:
Agree with you on this one. Maybe something we can address in a follow up PR?

Reply:
Agree with you both. Would be great if we could improve `showIf` to also include standard properties. Def work to a separate PR and probably a breaking change to the API.